### PR TITLE
Init widget

### PR DESCRIPTION
**Pull request by samasastudio**
created stateful Widget component and added unit test coverage for initial state and handleClick. 

@andyoe0102 @cmonez @krisdo

### CODE REVIEW

File: client/app.jsx [hunk 1]
```diff
@@ -21,9 +22,11 @@ class App extends React.Component {
   componentDidMount() {
 
     var listingId = this.state.listingId
```
Comment (cmonez) [inline client/app.jsx:24]: Very interesting how you don't define a state object initially, just setState within the componentDidMount. Any reason why you chose this pattern? I'm just curious.

File: client/components/Widget.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+class Widget extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      listingId: this.props.listingData.listingId
+    };
+  }
+
+  handleClick() {
```
Comment (cmonez) [inline client/components/Widget.jsx:12]: Whats the point of this handleClick function defintion? I see it gets used in your function, just a little unsure how it comes into play....

**Comment by andyoe0102** [comment]
looks good. For creating calendar in your widget, have you heard of react-calendar??
https://www.npmjs.com/package/react-calendar

File: client/components/Widget.jsx (hunk 2 above)
Comment (samasastudio) [reply client/components/Widget.jsx:12]: it's not doing anything yet!  it'll come into play when i start implementing a scroll-to parts of the page and may use a modal if I have time.  just haven't built it out yet.

File: client/app.jsx (hunk 1 above)
Comment (samasastudio) [reply client/app.jsx:24]: GitHub has hidden it from us!  those snakes.

**Comment by samasastudio** [comment]
whaaaat, thank you @andyoe0102 !!!

**Pull request merged**